### ISSUE TITLE
front container crashing after upgrading to 2.0.39

### ISSUE THREAD

**Issue by jkuester** [Mailu/Mailu#3193]
## Environment & Version

- Docker: `25.0.3, build 4debf41`
- Docker Compose: `v2.24.6`
- Mailu: `2.0.39`

## Description

My instance was stable on Mailu `2.0.38`.  I upgraded to `2.0.39` without changing any other configuration settings and now my `front` container is continuously crashing and restarting.

## Replication Steps

I am running Mailu behind Traefik. Traefik is bound to `80` and `443` on my host and proxies requests to my mailu `front` container.  Here are the Traefik labels I use for `front`:

```yml
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.mailu_front.rule=(Host(`${MAIL_HOST_NAME}`) || Host(`${IMAP_HOST_NAME}`) || Host(`${SMTP_HOST_NAME}`) || Host(`${EXCHANGE_HOST_NAME}`))"
      - "traefik.http.routers.mailu_front.tls.certresolver=acme_resolver"
      - "traefik.http.services.mailu_front.loadbalancer.server.port=80"
```

The rest of the necessary ports are bound directly to `front`

Also, I am using `TLS_FLAVOR=mail-letsencrypt`.

## Observed behaviour

- Update the configured version of Mailu from `2.0.38` to `2.0.39`.
- Run `docker compose up -d`
- All the containers are recreated
- The `front` container never starts successfully

## Logs

The `front` container logs just continuously repeat these lines every time it restarts:

```
front-1      | WARNING:root:Your CPU has Advanced Vector Extensions available, we recommend you enable hardened-malloc earlier in the boot process by adding LD_PRELOAD=/usr/lib/libhardened_malloc.so to your mailu.env
front-1      | nginx: [warn] "ssl_stapling" ignored, host not found in OCSP responder "r3.o.lencr.org" in the certificate "/certs/letsencrypt/live/mailu-ecdsa/nginx-chain.pem"
front-1      | Saving debug log to /var/log/letsencrypt/letsencrypt.log
front-1      | nginx: [warn] "ssl_stapling" ignored, host not found in OCSP responder "r3.o.lencr.org" in the certificate "/certs/letsencrypt/live/mailu/nginx-chain.pem"
front-1      | An unexpected error occurred:
front-1      | requests.exceptions.ConnectionError: HTTPSConnectionPool(host='acme-v02.api.letsencrypt.org', port=443): Max retries exceeded with url: /directory (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x72ed27754210>: Failed to establish a new connection: [Errno -3] Try again'))
front-1      | Ask for help or search for solutions at https://community.letsencrypt.org. See the logfile /var/log/letsencrypt/letsencrypt.log or re-run Certbot with -v for more details.
front-1      | Saving debug log to /var/log/letsencrypt/letsencrypt.log
front-1      | nginx: [emerg] host not found in upstream "imap" in /etc/nginx/nginx.conf:259
front-1 exited with code 0
```

## Additional Context

I imagine the issue is related to [these changes](https://github.com/Mailu/Mailu/compare/2.0.38...2.0.39#diff-4ba3b591630f4f7d4f123e191fe2423c5cf9739cc0d0e5ce24aa1841d1b81c6f) in the `nginx/config.py`, but I am not familiar enough with what is going on to be exactly sure of the root cause of the problem. 

As noted [on the Matrix channel](https://matrix.to/#/!RJFCFtixHgPhzacdhW:tedomum.net/$171059553323pQYxT:matrix.kuester7.com?via=ghostdub.de&via=matrix.org&via=tedomum.net) I guess I am not the only one experiencing this behavior!

**Comment by Diman0** [comment]
This error states that you were rate limited by the letsencrypt endpoint. As a result no new certificate was retrieved and mailu cannot start.

You could run `docker compose exec front cat /var/log/letsencrypt/letsencrypt.log` to check the log output from letsencrypt. It should show the reason why the ratelimit was triggered.

**Comment by jkuester** [comment]
Thanks for the reply!

For better or worse, when I spun everything back up (now a few hours later), I did not get the same LetsEncrypt error.  (In fact, the `front` container did not crash at all.)  Unfortunately, it seems the the HTTP/s traffic is still not getting proxyed though correctly as I am getting Gateway timeouts when trying to connect to the Webmail/Admin apps.  My current theory is that this proxy issue was probably the root cause of the LetsEncrypt issues and I just eventually got rate-limited trying to renew some certs...  

This is looking more and more like an issue with my proxy/Traefik setup.  I am going to close this issue and work on aligning my setup with the [recommended Traefik config](https://mailu.io/master/reverse.html#traefik-as-reverse-proxy) (which did not exist when I first deployed...).  Thanks again for having a look!
